packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -956,16 +956,14 @@ const alternateTypeKey = createStateSymbol("alternateType");
 export const $alternateType: AlternateTypeDecorator = (
   context: DecoratorContext,
   source: ModelProperty | Scalar,
-  alternate: Scalar,
+  alternate: Type,
   scope?: LanguageScopes,
 ) => {
-  if (source.kind === "ModelProperty" && source.type.kind !== "Scalar") {
+  if (source.kind === "Scalar" && alternate.kind !== "Scalar") {
```

Comment:
so, after design change, we only check the scalar now?

Reply:
> so, after design change, we only check the scalar now?

Yes.

Reply:
emmm, actually, i prefer to check if the alternate type kind is same with original type. then model to model, scalar to scalar, etc.

Reply:
@tadelesh 
Based on a [user requirement](https://github.com/Azure/typespec-azure/issues/1757#issuecomment-2465623277) of the original issue, the user would like to alternate the type from string to a custom type.
For `TypeProperty`, I think we can allow the user to alternate the type as they wantas garbage-in-garbage-out. Would it cause any side-effect as you can see?